core/protocols/aws-xml-protocol/src/main/java/software/amazon/awssdk/protocols/xml/internal/unmarshall/AwsXmlResponseTransformer.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.protocols.xml.internal.unmarshall;
+
+import static software.amazon.awssdk.awscore.util.AwsHeader.AWS_REQUEST_ID;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.awscore.AwsResponseMetadata;
+import software.amazon.awssdk.awscore.DefaultAwsResponseMetadata;
+import software.amazon.awssdk.core.SdkPojo;
+import software.amazon.awssdk.core.SdkStandardLogger;
+import software.amazon.awssdk.http.SdkHttpFullResponse;
+import software.amazon.awssdk.http.SdkHttpResponse;
+import software.amazon.awssdk.protocols.query.unmarshall.XmlElement;
+
+/**
+ * A transformer function that takes a parsed XML response and converts it into an {@link AwsResponse}. Used
+ * as a component in the {@link AwsXmlPredicatedResponseHandler}.
+ */
+@SdkInternalApi
+public final class AwsXmlResponseTransformer<T extends AwsResponse>
+        implements Function<AwsXmlUnmarshallingContext, T> {
+
+    private static final String X_AMZN_REQUEST_ID_HEADER = "x-amzn-RequestId";
```

Comment:
id-2 since they are common in xml?

Reply:
Still confused. Can you explain?

Reply:
id-2 and other extended response metadata is stored in `generateResponseMetadata` and accessible from service specific response metadata class, like `S3ResponseMetadata`

Reply:
As discussed with @spfink we're going to leave this as-is as it's existing/refactored code and there's no reason to touch it.